### PR TITLE
Add per-commit release job

### PR DESCRIPTION
**Pull request by howardjohn**
In T&R wg we decided to replace the daily builds with a per commit release job. This adds the script for this -- job to be added after this is merged

### CODE REVIEW

File: prow/release-commit.sh [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+export PATH=${GOPATH}/bin:${PATH}
+
+# cd to not impact go.mod
+(cd /tmp; go get istio.io/release-builder)
+
+release-builder build --manifest <(echo "${MANIFEST}")
+
+if [[ -z "${DRY_RUN:-}" ]]; then
+  release-builder publish --release "${WORK_DIR}/out" --gcsbucket "${GCS_BUCKET}" --dockerhub "${DOCKER_HUB}" --dockertags "${TAG},${NEXT_VERSION}-dev,latest"
+fi
```
Comment (nmittler) [inline prow/release-commit.sh]: nit: newline

File: prow/release-commit.sh [hunk 2]
```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+
+# Copyright 2018 Istio Authors
```
Comment (nmittler) [inline prow/release-commit.sh]: 2019?

**Comment by clarketm** [comment]
@rlenglet I just came across this PR in istio/test-infra#2000 that outline this particular issue. The issue is most likely the `build-tools` image. Furthermore, it is a *dirty* image so no guessing what it actually contains. I think the appropriate mitigation here is to replace/update the image for the appropriate `release-1.3` jobs.

**Pull request merged**